### DISCUSSION TITLE
Quartz as a  Service

### DISCUSSION THREAD

**Discussion by ColinA64** [quartznet/quartznet#987] [General]
Hi

I can  schedule jobs to both ramstore and database, but am having problems using Quartz as a service,  the install of the service seems fine,  attachments are testing the service when installed  with service running (issues)  not running seem ok I am new to Quartz so any help would be appreciated
![quartz_ServiceNotRunning](https://user-images.githubusercontent.com/69348790/95554960-fcaacc00-0a08-11eb-8cab-74e1a24167d3.PNG)
![Quartz_ServiceRunning](https://user-images.githubusercontent.com/69348790/95554967-fd436280-0a08-11eb-9ba0-647f206eb77c.PNG)

, TIA

**Comment by lahma** [comment]
So I presume that you need the remoting functionality to invoke scheduler remotely. This might be caused by some permission requirements that your principal running the Windows service is missing. Have you tried running the service with different permissions (different user), like network service?

If you don't need the remoting bits, you can just remove them from your configuration.

**Reply by ColinA64** [reply]
Hi thanks for the reply, currently the service is on the same  machine as the application running POC ,  have you working  examples of the config file for a service as not to sure of all the settings, thanks

**Reply by lahma** [reply]
Haven't seen you settings, but try removing `exporter.type` configuration if you don't need it.

**Comment by ColinA64** [comment]
Hi lahma

my config is 

```
# You can configure your scheduler in either <quartz> configuration section
# or in quartz properties file
# Configuration section has precedence

quartz.scheduler.instanceName = ServerScheduler
quartz.scheduler.instanceId = instance_one


# configure thread pool info
quartz.threadPool.type = Quartz.Simpl.SimpleThreadPool, Quartz
quartz.threadPool.threadCount = 10

# job initialization plugin handles our xml reading, without it defaults are used
quartz.plugin.xml.type = Quartz.Plugin.Xml.XMLSchedulingDataProcessorPlugin, Quartz.Plugins
quartz.plugin.xml.fileNames = ~/quartz_jobs.xml

# export this server to remoting context
# quartz.scheduler.exporter.type = Quartz.Simpl.RemotingSchedulerExporter, Quartz
quartz.scheduler.exporter.port = 555
quartz.scheduler.exporter.bindName = QuartzScheduler
quartz.scheduler.exporter.channelType = tcp
quartz.scheduler.exporter.channelName = httpQuartz

# this section configures the scheduler to use sql server ce as the jobstore
quartz.jobStore.type = Quartz.Impl.AdoJobStore.JobStoreTX, Quartz 
quartz.jobStore.useProperties = true
quartz.jobStore.dataSource = default
quartz.jobStore.tablePrefix = QRTZ_

#sql Server connection
quartz.dataSource.default.connectionString = Data Source=DBServer;Initial Catalog=Database;User Id=UserID;Password=UserPassword;
quartz.dataSource.default.provider = SSqlServer-20
quartz.jobStore.lockHandler.type = Quartz.Impl.AdoJobStore.UpdateLockRowSemaphore, Quartz
quartz.jobStore.driverDelegateType = Quartz.Impl.AdoJobStore.SqlServerDelegate, Quartz 
quartz.jobStore.clustered = true
```

I have commented out the exporter.type

**Reply by lahma** [reply]
You can comment out everything that starts with the `quartz.scheduler.exporter`, so four more lines. After that remoting shouldn't kick in (and you won't be able to use it).

**Comment by ColinA64** [comment]
Hi

Thanks for that getting no errors on starting service as ramstore, but when enabling  database  the service fais to start with  request did not respond in timely  fasion,  my config is:


quartz.scheduler.instanceName = ServerScheduler
quartz.scheduler.instanceId = testSceduler


quartz.threadPool.type = Quartz.Simpl.SimpleThreadPool, Quartz
quartz.threadPool.threadCount = 10


quartz.plugin.xml.type = Quartz.Plugin.Xml.XMLSchedulingDataProcessorPlugin, Quartz.Plugins
quartz.plugin.xml.fileNames = ~/quartz_jobs.xml



quartz.jobStore.type = Quartz.Impl.AdoJobStore.JobStoreTX, Quartz
quartz.jobStore.driverDelegateType = Quartz.Impl.AdoJobStore.SqlServerDelegate, Quartz
quartz.jobStore.useProperties = true
quartz.jobStore.dataSource = myDS
quartz.jobStore.tablePrefix = QRTZ_
quartz.jobStore.clustered = false
quartz.jobStore.lockHandler.type = Quartz.Impl.AdoJobStore.UpdateLockRowSemaphore, Quartz


quartz.dataSource.myDS.provider = SqlServer
quartz.dataSource.myDS.connectionString = Server=server;Database=database;User Id=user;Password=userpassword;

thanks for the help.

**Reply by lahma** [reply]
Check the logs.

**Comment by ColinA64** [comment]
Hi

getting a timeout was 30000 but have doubled it and still timing out

A timeout was reached (60000 milliseconds) while waiting for the Quartz Server service to connect. other apps can connect to database no problems

Tried service running as Local & Network service also as an account with admin rights same  for all  timeout
